volatility3/framework/plugins/windows/vadregexscan.py

```diff
@@ -0,0 +1,129 @@
+# This file is Copyright 2024 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import logging
+import re
+from typing import List
+
+from volatility3.framework import interfaces, renderers
+from volatility3.framework.configuration import requirements
+from volatility3.framework.interfaces import plugins
+from volatility3.framework.layers import scanners
+from volatility3.framework.renderers import format_hints
+from volatility3.plugins.windows import pslist, vadyarascan
+
+vollog = logging.getLogger(__name__)
+
+
+class VadRegExScan(plugins.PluginInterface):
+    """Scans all virtual memory areas for tasks using RegEx."""
+
+    _required_framework_version = (2, 0, 0)
+    _version = (1, 0, 0)
+    MAXSIZE_DEFAULT = 128
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        # Since we're calling the plugin, make sure we have the plugin's requirements
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Windows kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+            requirements.PluginRequirement(
+                name="pslist", plugin=pslist.PsList, version=(2, 0, 0)
+            ),
+            requirements.PluginRequirement(
+                name="vadyarascan", plugin=vadyarascan.VadYaraScan, version=(1, 0, 0)
+            ),
+            requirements.ListRequirement(
+                name="pid",
+                description="Filter on specific process IDs",
+                element_type=int,
+                optional=True,
+            ),
+            requirements.StringRequirement(
+                name="pattern", description="RegEx pattern", optional=False
+            ),
+            requirements.IntRequirement(
+                name="maxsize",
+                description="Maximum size in bytes for displayed context",
+                default=cls.MAXSIZE_DEFAULT,
+                optional=True,
+            ),
+        ]
+
+    def _generator(self, regex_pattern, procs):
+        regex_pattern = bytes(regex_pattern, "UTF-8")
+        vollog.debug(f"RegEx Pattern: {regex_pattern}")
+
+        for proc in procs:
+
+            # attempt to create a process layer for each proc
+            proc_layer_name = proc.add_process_layer()
+            if not proc_layer_name:
+                continue
+
+            # get the proc_layer object from the context
+            proc_layer = self.context.layers[proc_layer_name]
+
+            # get process sections for scanning
+            sections = vadyarascan.VadYaraScan.get_vad_maps(proc)
```

Comment:
This seems like an odd plugin for vad functionality to live in?  I'm not sure the function is worth being it's own function?  It seems a lot of other plugins just directly all `get_vad_root().traverse` and I'd probably recommend that rather than adding the requirement on a whole other plugin (particularly one that can likely do regex scanning itself?).  From a quick grep `get_vad_root` turns up in the following places:

```
windows/pe_symbols.py:            vad_root = proc.get_vad_root()
windows/vadinfo.py:        for vad in proc.get_vad_root().traverse():
windows/cmdscan.py:        for vad in conhost_proc.get_vad_root().traverse():
windows/suspicious_threads.py:            for vad in proc.get_vad_root().traverse():
windows/vadyarascan.py:        vad_root = task.get_vad_root()
windows/hollowprocesses.py:        for vad in proc.get_vad_root().traverse():
windows/svclist.py:        vad_root = proc.get_vad_root()
windows/malfind.py:        for vad in proc.get_vad_root().traverse():
windows/skeleton_key_check.py:        for vad in lsass_proc.get_vad_root().traverse():
windows/consoles.py:        for vad in conhost_proc.get_vad_root().traverse():
windows/dumpfiles.pywindows/pe_symbols.py:            vad_root = proc.get_vad_root()
windows/vadinfo.py:        for vad in proc.get_vad_root().traverse():
windows/cmdscan.py:        for vad in conhost_proc.get_vad_root().traverse():
windows/suspicious_threads.py:            for vad in proc.get_vad_root().traverse():
windows/vadyarascan.py:        vad_root = task.get_vad_root()
windows/hollowprocesses.py:        for vad in proc.get_vad_root().traverse():
windows/svclist.py:        vad_root = proc.get_vad_root()
windows/malfind.py:        for vad in proc.get_vad_root().traverse():
windows/skeleton_key_check.py:        for vad in lsass_proc.get_vad_root().traverse():
windows/consoles.py:        for vad in conhost_proc.get_vad_root().traverse():
windows/dumpfiles.pywindows/pe_symbols.py:            vad_root = proc.get_vad_root()
windows/vadinfo.py:        for vad in proc.get_vad_root().traverse():
windows/cmdscan.py:        for vad in conhost_proc.get_vad_root().traverse():
windows/suspicious_threads.py:            for vad in proc.get_vad_root().traverse():
windows/vadyarascan.py:        vad_root = task.get_vad_root()
windows/hollowprocesses.py:        for vad in proc.get_vad_root().traverse():
windows/svclist.py:        vad_root = proc.get_vad_root()
windows/malfind.py:        for vad in proc.get_vad_root().traverse():
windows/skeleton_key_check.py:        for vad in lsass_proc.get_vad_root().traverse():
windows/consoles.py:        for vad in conhost_proc.get_vad_root().traverse():
windows/dumpfiles.py:                for vad in proc.get_vad_root().traverse():
```

I think I'd just do it yourself and avoid pulling in a plugin just to do those 3 lines...

Reply:
That sounds fair! I opened the wrong plugin to get my inspiration... I'll sort it.

Reply:
Sorted that now - let me know what you think.